hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/CarbonitingMetadataFreezerActorSpec.scala

```diff
@@ -80,7 +81,141 @@ class CarbonitingMetadataFreezerActorSpec extends TestKitSuite("CarbonitedMetada
     // Finally, when the database responds appropriately, the actor should shut itself down:
     watch(actor)
     actor.underlyingActor.updateArchiveStatusPromise.success(1)
-    actor.stateName should be(Pending)
+    eventually {
+      actor.stateName should be(Pending)
+    }
+  }
+
+  it should "correctly handle failure received from IOActor" in {
+    val actor = TestFSMRef(new TestableCarbonitingMetadataFreezerActor(carboniterConfig, carboniteWorkerActor.ref, serviceRegistryActor.ref, ioActor.ref))
+    val workflowIdToFreeze = WorkflowId.randomId()
+
+    actor ! FreezeMetadata(workflowIdToFreeze)
+
+    serviceRegistryActor.expectMsgPF(10.seconds) {
+      case gma: GetMetadataAction => gma.workflowId should be(workflowIdToFreeze)
+    }
+    eventually {
+      actor.stateName should be(Fetching)
+    }
+
+    val jsonValue =
+      s"""{
+         |  "id": "$workflowIdToFreeze",
+         |  "status": "Successful"
+         |}""".stripMargin.parseJson
+
+    serviceRegistryActor.send(actor, SuccessfulMetadataJsonResponse(null, jsonValue.asJsObject))
+
+    var ioCommandPromise: Promise[Any] = null
+    ioActor.expectMsgPF(10.seconds) {
+      case command @ IoCommandWithPromise(ioCommand: IoWriteCommand, _) =>
+        ioCommand.file.pathAsString should be(HybridCarboniteConfig.pathForWorkflow(workflowIdToFreeze, "carbonite-test-bucket"))
+        ioCommand.content should be(jsonValue.prettyPrint)
+        ioCommand.compressPayload should be(true)
+        ioCommandPromise = command.promise
+    }
+    eventually {
+      actor.stateName should be(Freezing)
+    }
+
+    // Simulates the IoActor failing to complete the IO command successfully:
+    ioCommandPromise.failure(new RuntimeException("Cannot write metadata to GCS bucket"))
```

Comment:
That exception message is very real looking; in the chance that it ends up in a log it may be preferable to make clear it's an exception being intentionally generated from a test.